libs/video.liq

```diff
@@ -34,8 +34,38 @@ def video.add_image(~id="",~width=0,~height=0,~x=max_int(),~y=max_int(),~file,s)
   add([s,image])
 end
 
+# Input from ffmpeg.
+# @category Source / Video Processing
+# @param input ffmpeg options specifying the input
+def input.external.ffmpeg(~id="",input)
+  cmd = "ffmpeg #{input} -f avi -vf format=rgb24 -c:v rawvideo -r #{video.frame.rate()} -c:a pcm_s16le -ac 2 -ar #{audio.samplerate()} pipe:1"
+  (input.external.avi(cmd):source(audio=2,video=1,midi=0))
+end
+
 # ffmpeg's test source video (useful for testing and debugging).
 # @category Source / Video Processing
-def video.external.testsrc()
-  input.external.avi("ffmpeg -f lavfi -i testsrc -f avi -vf format=rgb24 -vcodec rawvideo -r #{video.frame.rate()} -s #{video.frame.width()}x#{video.frame.height()} pipe:1")
+def video.external.testsrc(~id="")
+  input.external.ffmpeg(id=id, "-f lavfi -i testsrc=size=#{video.frame.width()}x#{video.frame.height()}:rate=#{video.frame.rate()} -f lavfi -i sine=frequency=1000:sample_rate=#{audio.samplerate()}")
```

Comment:
This is gonna conflict with the external factorization PR. We should just merge it.